pkg/local_object_storage/engine/control.go

```diff
@@ -21,7 +21,9 @@ func (e *StorageEngine) open() error {
 
 	for id, sh := range e.shards {
 		if err := sh.Open(); err != nil {
-			return fmt.Errorf("open shard %s: %w", id, err)
+			if !e.cfg.isIgnoreMissingShards {
```

Comment:
with ignore flag set, we must not ignore failure. Instead, since we dont try to resurrect `sh` somehow, it must be removed from the `e.shards`

same for init